src/IlkRegistry.sol

```diff
@@ -137,10 +152,23 @@ contract IlkRegistry {
         (address _pip,) = spot.ilks(_ilk);
         require(_pip != address(0), "IlkRegistry/pip-invalid");
 
-        (address _flip,,) = cat.ilks(_ilk);
-        require(_flip != address(0), "IlkRegistry/flip-invalid");
-        require(FlipLike(_flip).cat() == address(cat), "IlkRegistry/flip-wrong-cat");
-        require(FlipLike(_flip).vat() == address(vat), "IlkRegistry/flip-wrong-vat");
+        (address _clip,,,) = dog.ilks(_ilk);
+
+        address  _xlip;
+        uint96  _class;
+        if (_clip != address(0)) {
+            require(ClipLike(_clip).dog() == address(dog), "IlkRegistry/clip-wrong-dog");
+            require(ClipLike(_clip).vat() == address(vat), "IlkRegistry/clip-wrong-vat");
+            _xlip = _clip;
+            _class = 1;
+        } else {
+            (address _flip,,)  = cat.ilks(_ilk);
+            require(_flip != address(0), "IlkRegistry/invalid-auction-contract");
+            require(FlipLike(_flip).cat() == address(cat), "IlkRegistry/flip-wrong-cat");
+            require(FlipLike(_flip).vat() == address(vat), "IlkRegistry/flip-wrong-vat");
+            _xlip = _flip;
+            _class = 2;
+        }
```

Comment:
```suggestion
        (address _xlip,,,) = dog.ilks(_ilk);

        uint96  _class = 1;
        if (_clip != address(0)) {
            require(ClipLike(_clip).dog() == address(dog), "IlkRegistry/clip-wrong-dog");
            require(ClipLike(_clip).vat() == address(vat), "IlkRegistry/clip-wrong-vat");
        } else {
            (address _xlip,,)  = cat.ilks(_ilk);
            require(_xlip != address(0), "IlkRegistry/invalid-auction-contract");
            require(FlipLike(_xlip).cat() == address(cat), "IlkRegistry/flip-wrong-cat");
            require(FlipLike(_xlip).vat() == address(vat), "IlkRegistry/flip-wrong-vat");
            _class = 2;
        }
```

Reply:
A bit shorter version, however yours is a bit more explicit. So feel free to take it or not.